src/Model/UserAction.php

```diff
@@ -174,6 +140,45 @@ public function execute(...$args)
         }
     }
 
+    protected function validate()
```

Comment:
good - but it checks if action is valid to be executed, not if valid like valid :)

so rename to `validateBeforeExecute` and 👍

Reply:
Done